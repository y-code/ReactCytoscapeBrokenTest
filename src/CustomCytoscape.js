import React, { useEffect, useRef } from "react";
import cytoscape from "cytoscape";
import Cytoscape from "react-cytoscapejs";

export function CustomCytoscape({ headless, elements }) {
  const layout = { name: "preset" };

  const cyRef = useRef();

  // Use cytoscape (non-react) if headless === true
  useEffect(() => {
    if (headless) {
      cyRef.current = cytoscape({
        container: document.getElementById("cyHeadless"),
        layout,
        elements
      });
    }
  }, [headless, elements, layout]);

  return headless ? (
    <div id="cyHeadless" />
  ) : (
    <Cytoscape
      layout={layout}
      cy={cy => {
        cyRef.current = cy;
      }}
      elements={elements}
      style={{ height: "300px", width: "300px" }}
    />
  );
}
