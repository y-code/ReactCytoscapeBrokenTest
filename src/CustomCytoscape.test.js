import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { CustomCytoscape } from "./CustomCytoscape";

describe("CustomCytoscape", () => {
  const elements = [];
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <CustomCytoscape elements={elements} headless={true} />,
      div
    );
  });

  it("matches snapshot", () => {
    const component = renderer.create(
      <CustomCytoscape elements={elements} headless={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
