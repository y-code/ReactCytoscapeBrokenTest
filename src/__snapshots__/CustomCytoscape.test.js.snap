// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCytoscape matches snapshot 1`] = `
<div
  id="cyHeadless"
/>
`;
